src/session/EnsureCookiePresentMiddleware.ts

```diff
@@ -15,13 +15,10 @@ const DEFAULT_REDIRECT_QUERY_PARAM_VALUE = "true";
  * present in the request
  */
 export function EnsureSessionCookiePresentMiddleware(config: CookiePresenceConfig): RequestHandler {
-    const { redirectQueryParameterName, redirectQueryParameterValue } = redirectionParametersSupplier(config);
+    const { redirectHeaderName, redirectHeaderValue } = redirectionParametersSupplier(config);
 
-    const hasPreviouslyBeenRedirected = previousRedirectPredicateFactory(redirectQueryParameterName,
-        redirectQueryParameterValue);
-
-    const constructQueryString = queryStringSupplierFactory(redirectQueryParameterName,
-        redirectQueryParameterValue);
+    const hasPreviouslyBeenRedirected = previousRedirectPredicateFactory(redirectHeaderName,
+        redirectHeaderValue);
 
     return (req: Request, res: Response, next: NextFunction) => {
         if (!Object.keys(req.cookies).includes(config.cookieName)) {
```

Comment:
do we need an `else` condition to remove the header once we are happy we have the session cookie present?

Reply:
Ahh yes, that would be the safest option. I'll put a change up and come back to you

Reply:
Presumably this would only be an issue if we just blanketly copy the request headers to the response, does this happen? The handler does not modify the outgoing request so hard to see what removing it from the request will acheive?

There is not really an API for doing this either so not sure how easy it will be to remove it from the request?

Reply:
I have updated with what I think you have suggested - more belts and braces but I suppose that will suffice?

Reply:
ah, i see your point. my assumption was that if this header is set, then it will be present for all subsequent request/response cycles hence the need to remove it once it has achieved its purpose.

from your browser tests so far, is it not persisting?

if it is, we might want to remove it..

so, something like:

``` 
...

} else {
    if (req.headers["x-header-name"]) {
        res.removeHeader["x-header-name"];
     }
}
```